test/controllers/class_controller_test.exs

```diff
@@ -253,4 +253,28 @@ defmodule CoursePlanner.ClassControllerTest do
     soft_deleted_course = Repo.get(Class, class.id)
     assert soft_deleted_course.deleted_at
   end
+
+  test "creates class and all attendance", %{conn: conn} do
+    course = insert(:course)
+    term1 = insert(:term, %{
+                            start_date: %Ecto.Date{day: 1, month: 1, year: 2017},
+                            end_date: %Ecto.Date{day: 1, month: 6, year: 2017},
+                            courses: [course]
+                           })
+
+    students = insert_list(3 ,:student)
+    offered_course = insert(:offered_course, %{term: term1, course: course, students: students})
+    class_attrs = %{@valid_attrs | offered_course_id: offered_course.id, status: "Active"}
+
+    conn = post conn, class_path(conn, :create), class: class_attrs
+    assert redirected_to(conn) == class_path(conn, :index)
+    assert Repo.get_by(Class, class_attrs)
+
+    class = List.first(ClassHelper.all_none_deleted())
+    attendances = AttendanceHelper.get_class_attendance_info(class.id)
+
+    Enum.map(students, fn(item_student) ->
```

Comment:
I am checking that the students that are having attendance are the exact ones attendance been created for them